guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/helpers/HttpGetHelper.java

```diff
@@ -46,35 +46,37 @@
     @Inject
     private GuvnorM2Repository repository;
 
-    public void handle( final HttpServletRequest request,
-                        final HttpServletResponse response,
-                        final ServletContext context ) throws IOException {
+    public void handle(final HttpServletRequest request,
+                       final HttpServletResponse response,
+                       final ServletContext context) throws IOException {
         String requestedFile = request.getPathInfo();
 
-        if ( requestedFile == null ) {
-            response.sendError( HttpServletResponse.SC_NOT_FOUND );
+        if (requestedFile == null) {
+            response.sendError(HttpServletResponse.SC_NOT_FOUND);
             return;
         }
 
-        requestedFile = URLDecoder.decode( requestedFile,
-                                           "UTF-8" );
+        requestedFile = URLDecoder.decode(requestedFile,
+                                          "UTF-8");
+
+        String repositoryName = request.getParameter("repository");
```

Comment:
@manstis yes, this is guaranteed, because the service that retrieves an artifact uses the default repository if it receives a null o empty string repository name. I tried not to change any default behavior.